lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,113 @@
 #!/bin/bash
 
-pushd "$(dirname "$0")" || exit
-mkdir -p "compiled_v0"
+V1_COMPILER_BUILD_PATH_VAR="V1_COMPILER_BUILD_PATH"
+V2_COMPILER_BUILD_PATH_VAR="V2_COMPILER_BUILD_PATH"
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+V0_CONTRACT_PATH="src_v0"
+V1_CONTRACT_PATH="src_v1"
+V2_CONTRACT_PATH="src_v2"
+
+V0_ARTIFACT_PATH="compiled_v0"
+V1_ARTIFACT_PATH="compiled_v1"
+V2_ARTIFACT_PATH="compiled_v2"
+
+V1_COMPILER_VERSION="1.1.1"
+V2_COMPILER_VERSION="2.2.0"
+
+V1_COMPILER_BIN_PATH="compilers/v1"
+V2_COMPILER_BIN_PATH="compilers/v2"
+
+fetch_compilers() {
+  local VERSION=$1
+  local OUT_DIR=$2
+
+  local VERSION_SHORT=${VERSION:0:1}
+  [ "$VERSION_SHORT" == "1" ] && local COMPILER_BUILD_PATH_VAR="$V1_COMPILER_BUILD_PATH_VAR"
+  [ "$VERSION_SHORT" == "2" ] && local COMPILER_BUILD_PATH_VAR="$V2_COMPILER_BUILD_PATH_VAR"
+  local COMPILER_BUILD_PATH="${!COMPILER_BUILD_PATH_VAR}"
+
+  echo "Fetching compiler v$VERSION binaries..."
+
+  mkdir -p "$OUT_DIR"
+
+  OS=$(uname)
+  ARCH=$(uname -m)
+
+  if [ "$OS" == "Linux" -a "$ARCH" == "x86_64" ]; then
```

Comment:
```suggestion
  if [ "$OS" == "Linux" && "$ARCH" == "x86_64" ]; then
```

`-a` is rather poorly defined, it is a good practice to use `&&` instead. More rationale -> https://www.shellcheck.net/wiki/SC2166

Reply:
the same goes for all occurences of `-a` in this script